esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -6268,119 +6212,194 @@ void CWsDfuEx::exportRecordDefinitionBinaryType(const char *recordDefinition, Me
 
-        StringBuffer jobId = requestBase.getJobId();
-        if (jobId.isEmpty())
-            jobId.appendf("Create %s on %s", tempFileName.str(), clusterName);
+        if (requestId.isEmpty())
+            requestId.appendf("Create %s on %s", normalizedFileName.str(), clusterName);
 
-        Owned<IFileDescriptor> fileDesc = createFileDescriptor(tempFileName, clusterTypeEx, groupName, group);
-        fileDesc->queryProperties().setProp("@job", jobId);
+        Owned<IFileDescriptor> fileDesc = createFileDescriptor(tempFileName, clusterTypeString(clusterType, false), groupName, group);
+        fileDesc->queryProperties().setProp("@job", requestId);
         if (!userId.isEmpty())
             fileDesc->queryProperties().setProp("@owner", userId);
+        fileDesc->queryProperties().setProp("ECL", recordDefinition);
 
-        Owned<IDistributedFile> file = queryDistributedFileDirectory().createNew(fileDesc);
-        file->setAccessed();
-        file->setECL(recordDefinition);
-        file->queryAttributes().setPropBin("_rtlType", layoutBin.length(), layoutBin.toByteArray());
-        file->queryAttributes().setPropInt64("@recordCount", 0);
-        file->queryAttributes().setPropInt64("@size", 0);
+        MemoryBuffer layoutBin;
+        exportRecordDefinitionBinaryType(recordDefinition, layoutBin);
+        if (0 == layoutBin.length())
+            throw makeStringExceptionV(ECLWATCH_INVALID_ECLRECDEF, "DFUCreateFile: Failed to parse ECL record definition: %s.", recordDefinition);
+        fileDesc->queryProperties().setPropBin("_rtlType", layoutBin.length(), layoutBin.toByteArray());
 
-        getFileAccessBeforePublish(context, tempFileName, clusterName, jobId, layoutBin, fileDesc, userDesc, requestBase, resp);
+        // NB: if file has copies on >1 cluster, they must share the same key
+        StringBuffer keyPairName;
+        unsigned port;
+        bool secure;
 
-        //create FileId
-        StringBuffer fileID;
-        fileID.set(groupName.str()).append(DFUFileIdSeparator).append(clusterName).append(DFUFileIdSeparator).append(tempFileName.str());
-        resp.setFileId(fileID.str());
+        std::vector<std::string> groups;
+        groups.push_back(groupName.str());
+        getFileDafilesrvConfiguration(keyPairName, port, secure, normalizedFileName, groups);
+
+        Owned<IPropertyTree> metaInfo = createDFUFileMetaInfo(tempFileName, fileDesc, requestId, "WRITE", expirySecs, userDesc, keyPairName, port, secure, maxFileAccessExpirySeconds);
+        metaInfo->setProp("clusterName", clusterName);
+
+        StringBuffer metaInfoBlob;
+        encodeDFUFileMeta(metaInfoBlob, metaInfo, env);
+
+        IEspDFUFileAccessInfo &accessInfo = resp.updateAccessInfo();
+        accessInfo.setMetaInfoBlob(metaInfoBlob);
+        if (returnTextResponse)
+        {
+            getFileAccessInfo(context, *fileDesc, true, accessInfo);
+            accessInfo.setExpiryTime(metaInfo->queryProp("keyPairName"));
```

Comment:
noticed the same oddity above, perhaps this is intentional, but the "keyPairName" prop doesn't seem to be directly correlated to the expirytime.

Reply:
yep, will fix.